README.md

```diff
@@ -11,19 +11,583 @@ JavaScript this Code-Along
+ways.
 
-This will lead to a bunch of contrived examples, but I think it's really important for students to get the hang of how `this` works.
+Don't forget to code along in your console!
 
-The main point is that students need a ton of practice with `call()` and `apply()`
+## Plain Ol' JavaScript Objects (POJOs)
 
-`bind()` is a tricky sibling of `call()` and `apply()` in that it returns a new function. This might trip students up, so it's worth giving them a bit of practice. It's worth introducing in the context of, e.g., `setTimeout()` and `setInterval()` (which students should know about by this point).
+We can create objects in JavaScript to associate values to properties,
+like a sandwich:
+
+```js
+var pbj = {
+  bread: "white",
+  ingredients: ["peanut butter", "jelly"],
+  cut: "triangles"
+}
+```
+
+We can also use a constructor function to create all kinds of sandwich
+objects:
+
+```js
+function Sandwich(bread, ingredients, cut) {
+  this.bread = bread;
+  this.ingredients = ingredients;
+  this.cut = cut;
+}
+
+var blt = new Sandwich("white", ["bacon","lettuce","tomato","mayo"],"rectangle");
+var reuben = new Sandwich("rye", ["corned beef","sauerkraut","swiss","russian dressing"],"diagonal");
+```
+
+And we can even attach a function to an object like this:
+
+```js
+var pbj = {
+  bread: "white",
+  ingredients: ["peanut butter", "jelly"],
+  cut: "triangles",
+  name: "peanut butter and jelly",
+  serve: function() {
+    console.log("here's your " + this.name + ", enjoy!");
+  }
+}
+```
+
+Now try calling `pbj.serve()`.
+
+### this and Functions
+
+When we called `pbj.serve()` above, it gave us the message with the
+`name` value for our sandwich. The `this.name` refers to the `name` of
+the `pbj` object, because `pbj` invoked the `serve()` function, becoming
+the owner of that function.
+
+If we wanted to explicitly construct a BLT with a `serve()` function, we
+could do that:
+
+```js
+var blt = {
+  bread: "white",
+  ingredients: ["bacon", "lettuce", "tomato"],
+  cut: "diagonal",
+  name: "BLT",
+  serve: function() {
+    console.log("here's your " + this.name + ", enjoy!");
+  }
+}
+
+blt.serve();
+```
+
+This works, but now we're repeating this `serve` function everywhere and
+that's not only a violation of the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) but also just a total pain.
+
+How can we apply what we know about constructor functions to this
+problem?
+
+It stands to reason that if we can set primitive property values in a
+constructor function, we should be able to create methods on the object
+we're constructing as well. Let's give it a shot.
+
+```js
+function Sandwich(bread, ingredients, cut) {
+  this.bread = bread;
+  this.ingredients = ingredients;
+  this.cut = cut;
+  this.serve = function() {
+    console.log("here's your " + this.name + ", enjoy!");
+  }
+}
+
+var blt = new Sandwich("white", ["bacon","lettuce","tomato","mayo"],"rectangle");
+
+blt.serve();
```

Comment:
This is great! I really like how you walk students through to this obvious (but wrong) implementation.